Core/PaymentMethodHelper.php

```diff
@@ -80,7 +82,143 @@ public static function getModulePayments()
      */
     public static function getMandate($iOrderNumber)
     {
-        $iLength = self::MAX_MANDATE_ID_LENGTH - 1 - strlen((string) time());
-        return new Mandate(substr($iOrderNumber, 0, $iLength) . '-' . time());
+        $sTime = (string) time();
+        $iLength = self::MAX_MANDATE_ID_LENGTH - 1 - strlen($sTime);
+        return new Mandate(substr($iOrderNumber, 0, $iLength) . '-' . $sTime);
+    }
+
+    /**
+     * Returns account holder for SEPA Direct Debit
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getAccountHolder()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['accountHolder'];
+    }
+
+
+    /**
+     * Returns IBAN
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getIban()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['iban'];
+    }
+
+    /**
+     * Returns BIC
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public function getBic()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['bic'];
+    }
+
+    /**
+     * Generates SEPA mandate html body
+     *
+     * @param Basket $oBasket
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getSepaMandateHtml($oBasket)
+    {
+        $oShop = Helper::getShop();
+        $iOrderNumber = Helper::getSessionChallenge();
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        $sSepaMandateHeader = self::getSepaMandateHeader($oShop, $oPayment, $iOrderNumber);
+        $sSepaMandateFooter = self::getSepaMandateFooter($oShop);
+        $sSepaMandateMain = self::getSepaMandateMainText($oShop);
+
+        if ($oPayment->oxpayments__wdoxidee_sepamandatecustom->value) {
+            $sSepaMandateMain = $oPayment->oxpayments__wdoxidee_sepamandatecustom->value;
+        }
+
+        return $sSepaMandateHeader . $sSepaMandateMain . $sSepaMandateFooter;
+    }
+
+    /**
+     * Generates SEPA mandate header text
+     *
+     * @param Shop    $oShop
+     *
+     * @param Payment $oPayment
+     *
+     * @param integer $iOrderNumber
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getSepaMandateHeader($oShop, $oPayment, $iOrderNumber)
+    {
+        return '<h3>' . Helper::translate('wd_sepa_mandate') . '</h3><hr>
+            <i>' . Helper::translate('wd_creditor') . '</i><p style="margin-bottom: 30px">' .
+            $oShop->oxshops__oxfname . ' ' . $oShop->oxshops__oxlname . ',<br>' .
+            $oShop->oxshops__oxstreet . '<br>' .
```

Comment:
Should we add `<br>` tags in case address data is empty? This is not only about this line, but about the html construction in general.